tests/unit/test_client_base.py

```diff
@@ -351,3 +337,61 @@ def test_vault_client_move_secrets_overwrite_force(vault):
     vault.move_secrets("a", "b", force=True)
 
     assert vault.db == {"b": "c"}
+
+
+def test_vault_client_base_render_template(vault):
+
+    vault.db = {"a/b": "c"}
+
+    assert vault.render_template("Hello {{ vault('a/b') }}") == "Hello c"
+
+
+def test_vault_client_base_render_template_path_not_found(vault):
+    with pytest.raises(exceptions.VaultRenderTemplateError):
+        vault.render_template("Hello {{ vault('a/b') }}")
+
+
+@pytest.mark.parametrize(
+    "vault_contents, expected",
+    [
+        # Secret is not a template
+        ({"a": "b"}, "b"),
+        # Secret not a string
+        ({"a": ["yay"]}, ["yay"]),
+        # Secret is a template without variable expansion
+        ({"a": "!template!b", "b": "c"}, "b"),
+        # Secret is a template
+        ({"a": "!template!{{ vault('b') }}", "b": "c"}, "c"),
```

Comment:
what happens when `"b":  {"u": "x", "p": "z"}` ?
is it possible to access "u" and "p" values (ie. `"!template!{{ vault('b').u }}"` ?

Reply:
here is an example :
```
$ vault set --yaml --stdin composite
username: foo
password: bar
Done
$ vault get composite
---
password: bar
username: foo
$ vault set username '!template!username={{ vault("composite")["username"] }}'         
Done
$ vault get --text username
username=foo
```

Reply:
ok nice then

Reply:
Can we test this then ?